Elaborate more on the route priority order (#3374)
withastro/docs: src/content/docs/en/core-concepts/routing.mdx

</FileTree>

Astro needs to know which route should be used to build the page. To do so, it sorts them according to the following rules:
1- Routes without path parameters
- Endpoints without path parameters take precedence over all other routes
- Routes without path parameters take precedence over dynamic routes

2- Dynamic routes with path parameters
- Static endpoints using named parameters take precedence over other static routes
- Static routes using named parameters take precedence over server endpoints using named parameters
- Server endpoints using named parameters take precedence over server routes using named parameters
- Server routes using named parameters take precedence over other routes using rest parameters

3- Dynamic routes with rest parameters
- Rest parameters have the lowest priority
- Static endpoints using rest parameters take precedence over other static routes using rest parameters
- Static routes using rest parameters take precedence over server endpoints using rest parameters
- Server endpoints using rest parameters take precedence over server routes using rest parameters

> - Server routes using rest parameters take precedence over other routes using rest parameters

Author: I wonder if it would be a good idea to create a [priority queue](https://www.google.com/search?q=Priority+Queue&tbm=isch) or a pyramid component to represent these

- Static routes without path parameters will take precedence over all other routes
- Dynamic routes using named parameters take precedence over rest parameters
- Pre-rendered dynamic routes take precedence over server dynamic routes
- Rest parameters have the lowest priority
- Endpoints always take precedence over pages

> - Ties are resolved alphabetically

Author: Just bringing back the line break
```suggestion
- Ties are resolved alphabetically

```

> - Rest parameters have the lowest priority
> - Ties are resolved alphabetically
> 
> sMoreover, endpoints always take precedence over pages. Also, if you have dynamic routes (using rest parameters or named parameters) that are prerendered, they will take precedence too over dynamic routes that aren't.
> 

Reviewer: ```suggestion
- Pre-rendered dynamic routes take precedence over static dynamic routes
- Rest parameters have the lowest priority
- Endpoints always take precedence over pages
- Ties are resolved alphabetically
```
What do you think about just adding to our list, like this? (view it with the points in the lines just above)

Author: That works for me, thanks for suggesting that!
There's just a minor change to do, in the first bullet point in “over static dynamic routes” it should be “**server**” instead of “**static**”.

Reviewer: Perfect, thanks!
